src/types/chain.ts

```diff
@@ -34,6 +34,8 @@ export type ChainConstants = {
   }
   /** Collection of contracts */
   contracts?: {
+    [key: string]: ChainContract | { [chainId: number]: ChainContract }
+  } & {
```

Comment:
Should be like this, right?

```suggestion
    [key: string]: ChainContract | { [chainId: number]: { [key: string]: ChainContract } }
  } & {
```

Reply:
The named part is already at L37:5-17

Reply:
Yeah the format is {"optimismPortal": {8543: "0x..."}}

I'm not sure we need the "deployed at" info but I guess good to stay consistent.

Reply:
Ofc we _could_ also do it as chainId>name>address, instead